sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_content_understanding_content_analyzers_operations_async.py

```diff
@@ -0,0 +1,762 @@
+    """Delete an analyzer and assert it was deleted successfully.
+
+    Args:
+        client: The ContentUnderstandingClient instance
+        analyzer_id: The analyzer ID to delete
+        created_analyzer: Whether the analyzer was created (to determine if cleanup is needed)
+
+    Raises:
+        AssertionError: If the analyzer still exists after deletion
+    """
+    if created_analyzer:
+        print(f"Cleaning up analyzer {analyzer_id}")
+        try:
+            await client.content_analyzers.delete(analyzer_id=analyzer_id)
+            # Verify deletion
+            assert not await analyzer_in_list_async(
+                client, analyzer_id
+            ), f"Deleted analyzer with ID '{analyzer_id}' was found in the list"
+            print(f"Analyzer {analyzer_id} is deleted successfully")
+        except Exception as e:
+            # If deletion fails, the test should fail
+            raise AssertionError(f"Failed to delete analyzer {analyzer_id}: {e}") from e
+    else:
+        print(f"Analyzer {analyzer_id} was not created, no cleanup needed")
+
+
+async def download_keyframes_and_assert_async(
+    client, analysis_operation_id: str, result, test_py_file_dir: str, identifier: Optional[str] = None
+) -> None:
+    """Download keyframes from video analysis result and assert their existence (async version).
+
+    Downloads up to 3 keyframes: first, middle, and last frame to avoid duplicates.
+
+    Args:
+        client: The ContentUnderstandingClient instance
+        analysis_operation_id: The operation ID from the analysis
+        result: The analysis result containing markdown with keyframes
+        test_py_file_dir: The directory where pytest files are located
+        identifier: Optional unique identifier to avoid conflicts (e.g., analyzer_id)
+
+    Returns:
+        None
+
+    Raises:
+        AssertionError: If no keyframes are found in the analysis result
+    """
+    keyframe_ids = set()
+
+    # Iterate over contents to find keyframes from markdown
+    for content in result.contents:
+        # Extract keyframe IDs from "markdown" if it exists and is a string
+        markdown_content = getattr(content, "markdown", "")
+        if isinstance(markdown_content, str):
+            # Use the same regex pattern as the official sample: (keyFrame\.d+)\.jpg
+            keyframe_ids.update(re.findall(r"(keyFrame\.\d+)\.jpg", markdown_content))
+
+    print(f"Found keyframe IDs in markdown: {keyframe_ids}")
+
+    # Assert that keyframe IDs were found in the video analysis
+    assert (
+        keyframe_ids
+    ), "No keyframe IDs were found in the video analysis markdown content. Video analysis should generate keyframes that can be extracted using regex pattern."
+
+    print(f"Successfully extracted {len(keyframe_ids)} keyframe IDs from video analysis")
+
+    # Sort keyframes by frame number to get first, middle, and last
```

Comment:
Overkill?  Perhaps just take the first, middle, and last frame based on order in markdown?